src/inspect_ai/_view/www/src/client/api/api-view-server.ts

```diff
@@ -8,354 +8,415 @@ import {
   SampleDataResponse,
 } from "./types";
 
-/* global __VIEW_SERVER_API_URL__ */
-const API_BASE_URL = __VIEW_SERVER_API_URL__ || "";
-const loaded_time = Date.now();
-let last_eval_time = 0;
-
-function buildApiUrl(path: string): string {
-  if (!API_BASE_URL) {
-    return path;
-  }
-  const base = API_BASE_URL.endsWith("/")
-    ? API_BASE_URL.slice(0, -1)
-    : API_BASE_URL;
-  const cleanPath = path.startsWith("/") ? path : `/${path}`;
-  return base + cleanPath;
-}
+const LOADED_TIME = Date.now();
 
-function isApiCrossOrigin(): boolean {
-  try {
-    return Boolean(
-      API_BASE_URL && new URL(API_BASE_URL).origin !== window.location.origin,
-    );
-  } catch (TypeError) {
-    return false;
-  }
-}
+let lastEvalTime = 0;
 
-async function client_events() {
-  const params = new URLSearchParams();
-  params.append("loaded_time", String(loaded_time.valueOf()));
-  params.append("last_eval_time", String(last_eval_time.valueOf()));
-  return (await api("GET", `/events?${params.toString()}`)).parsed;
-}
+type HeaderProvider = () => Promise<Record<string, string>>;
 
-async function eval_log(
-  file: string,
-  headerOnly?: number,
-  _capabilities?: Capabilities,
-): Promise<LogContents> {
-  return await api(
-    "GET",
-    `/logs/${encodeURIComponent(file)}?header-only=${headerOnly}`,
-  );
-}
+let globalHeaderProvider: HeaderProvider | undefined = undefined;
```

Comment:
Now that `createViewServerApi` accepts a `headerApiProvider`, we should remove the global altogether and just pass the provider to `createServerFetchApi` directly.